astro/src/content/docs/extend/code/lambdas/testing.mdx

```diff
@@ -0,0 +1,429 @@
+
+Make a new file called `userCreator.js` in the app folder and paste the following code into it.
+
+<RemoteCode lang="javascript" url="https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/userCreator.js" />
+
+The code above has two functions:
+
+- The `createRandomUser` function, which creates a User request object with `const request` and then sends it to `fusion.register()`. Details on this can be found in the [TypeScript client library interface](https://github.com/FusionAuth/fusionauth-typescript-client/blob/master/src/FusionAuthClient.ts).
+- The `deleteUser` function, which you can use in the tests to delete the user just created.
+
+Run the code to test user creation with the following command.
+
+```bash
+node userCreator.js
+```
+
+In FusionAuth, click on "Users" to check that a new user called `lambdatestuser` has been created. You can delete the `createRandomUser(uuidv4());` line in `userCreator.js` as each test will use a new temporary user. This will allow you to add multiple lambda tests while avoiding potential conflicts between test users and permissions.
+
+#### Write The Test
+
+Now you will test that the lambda returns "Goodbye World", which will confirm that the CLI `update` command worked.
+
+Create a file called `userLogin.js` and add the following code.
+
+<RemoteCode lang="javascript" url="https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/userLogin.js" />
+
+This helper file allows your tests to log in to FusionAuth programmatically. The `login` function calls the FusionAuth TypeScript library. It then decodes the JWT response and returns its `message` property.
+
+Now create a test file that will use it, `test_1.js`, and add the following code.
+
+<RemoteCode lang="javascript" url="https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/test_1.js" />
+
+This file starts with a declaration of constant variables that match the `kickstart.json` and `.env` files. The `login` function is called by the `tape` function `test`. This test specifies the name of the test, says that it expects exactly one assertion to occur with `plan`, checks that calling `login` returns the property you expect from the lambda updated earlier, and exits. Even if the test fails, the `finally` clause will delete the temporary user created.
+
+Run it with the following command.
+
+```bash
+node test_1.js
+```
+
+The output should be as follows.
+
+```bash
+TAP version 13
+# test login returns JWT with "Goodbye World"
+User c82aced2-b25b-4390-a4a2-72562b9bc13b created successfully
+ok 1 should be truthy
+User c82aced2-b25b-4390-a4a2-72562b9bc13b deleted successfully
+
+1..1
+# tests 1
+# pass  1
+
+# ok
+```
```

Comment:
I got

TAP version 13
# test login returns JWT with "Goodbye World"
User 85bf89dd-4f26-408e-be99-305b90c73f0b created successfully
User 84d7e42b-8668-401c-bb4a-68fa5f7b6801 created successfully
User 85bf89dd-4f26-408e-be99-305b90c73f0b deleted successfully
not ok 1 TypeError: login is not a function
  ---
    operator: error
    stack: |-
      TypeError: login is not a function
          at Test.<anonymous> (/Users/mark.robustelli/Projects/doc-testing-lambdas/fusionauth-example-testing-lambdas/app/test_1.js:15:26)
          at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  ...

1..1
# tests 1
# pass  0
# fail  1

Reply:
This works fine here. What is your Node version?
![image](https://github.com/FusionAuth/fusionauth-site/assets/1877191/3632ecb1-a661-498b-8fba-b9e09bdf5f1c)